### ISSUE TITLE
[New Rule] Use `Path`s for reading and writing files

### ISSUE THREAD

**Issue by TomFryers** [MartinThoma/flake8-simplify#111]
## Explanation

Using `Path`s requires much less code, particularly if `Path`s are already being used to represent `Path`s. The necessary methods were added in Python 3.5, so shouldn’t be a backwards-compatibility concern.

## Example

```python
# Bad
with open("foo.txt") as f:
    text = f.read()
with open("bar.txt") as f:
    f.write(text)

with open("baz.png", "rb") as f:
    data = f.read()
with open("foobar.png", "wb") as f:
    f.write(data)

# Good
from pathlib import Path

text = Path("foo.txt").read_text()
Path("bar.txt").write_text(text)

data = Path("baz.png").read_bytes()
Path("foobar.png").write_bytes(data)
```

**Comment by gladykov** [comment]
There is a separate plugin just for that: https://gitlab.com/RoPP/flake8-use-pathlib
